File: 🟡 Medium/2285 - Maximum Total Importance of Roads/_2285MaxTotalImpOfRoads.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        long[] freq = new long[n];

        for (int[] edge : roads) {
            freq[edge[0]]++;
            freq[edge[1]]++;
        }

        Arrays.sort(freq);

        long imp = 0;
        long k = 1;

        for (long f : freq) {
            imp += (k * f);
            k++;
        }

        return imp;

    }
}

File: 🟡 Medium/2285 - Maximum Total Importance of Roads/_2285MaxTotalImpOfRoads.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        freq = [0] * n

        for edge in roads:
            freq[edge[0]] += 1
            freq[edge[1]] += 1

        freq.sort()

        imp = 0
        k = 1

        for f in freq:
            imp += k * f
            k += 1

        return imp
